src/builtins/hmr-runtime.js

```diff
@@ -15,7 +15,7 @@ function Module() {
 module.bundle.Module = Module;
 
 if (!module.bundle.parent && typeof WebSocket !== 'undefined') {
-  var ws = new WebSocket('ws://' + window.location.hostname + ':{{HMR_PORT}}/');
+  var ws = new WebSocket('ws://' + location.hostname + ':{{HMR_PORT}}/');
```

Comment:
Could we use `self.location.hostname` to avoid the implicit global? I think this will work in both contexts.

Reply:
I tried that, but then the tests throw `ReferenceError: self is not defined`, however I found out that I can reference `location` from the `global` object, which you define on the top level, I'll use that.